### PR TITLE
use vec! to initialize vec with None values

### PR DESCRIPTION
**Pull request by AxelNennker**
Signed-off-by: Axel Nennker <axel.nennker@telekom.de>

### CODE REVIEW

File: libindy/src/services/pool/networker.rs [hunk 1]
```diff
@@ -190,9 +190,7 @@ impl PoolConnection {
             });
         }
 
-        let mut sockets: Vec<Option<ZSocket>> = Vec::new();
-
-        for _ in 0..nodes.len() { sockets.push(None); }
+        let mut sockets: Vec<Option<ZSocket>> = vec![None; nodes.len()];
```
Comment (jovfer) [inline libindy/src/services/pool/networker.rs]: it fails compilation on Jenkins https://build.sovrin.org/job/indy-sdk/job/indy-sdk-ci/job/PR-1755/1/execution/node/52/log/

Comment (AxelNennker) [reply libindy/src/services/pool/networker.rs]: Sorry, it seems I forgot to follow my usual procedure I do before sending PRs. I'll close this.

Comment (jovfer) [reply libindy/src/services/pool/networker.rs]: np. suggested changes was so obvious for me that I was totally surprised to have a compilation error when I wrote this code about year ago. And when you submitted a PR yesterday I was believing that behavior of the rust compiler had been fixed in one of the newer version

Comment (AxelNennker) [reply libindy/src/services/pool/networker.rs]: No, it is not fixed.
https://users.rust-lang.org/t/how-to-initialize-vec-option-t-with-none/30580/12

**Pull request closed without merging**